app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -197,6 +195,12 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         }
 
         captureSnapshotTelemetryMetrics()
+
+        setupAppAllStartTelemetry()
+    }
+
+    protected open fun setupAppAllStartTelemetry() {
+        appAllSourceStartTelemetry = AppAllSourceStartTelemetry(intent, components.analytics.metrics, false)
```

Comment:
What if the app is backgrounded and HomeActivity needs to be recreated? We need to ensure we're checking backgrounded here too, right? This is essentially the same case as `onNewIntent`, right? Can we just use the same function for both?

Also this doesn't cover the case that HomeActivity isn't launched at all. Better yet, could we use `ProcessLifecycleOwner.onStart` for this?

Reply:
If the app is backgrounded, and HomeActivity is recreated, `setupAppAllStartTelemetry` would get created again since I'm calling `setupAppAllStartTelemetry()` on HomeActivity's onCreate method. which would ensure that we are still tracking the event?

Im not sure what you mean by 
> This is essentially the same case as onNewIntent, right? Can we just use the same function for both?

The `onNewIntent` is not called when activity is first launched. 

What would be a case where `HomeActivity` is not launched?

Reply:
> Better yet, could we use ProcessLifecycleOwner.onStart for this?

I think my other comments are irrelevant if we use `ProcessLifecycleOwner` instead of attaching to `HomeActivity` directly.

> What if the app is backgrounded and HomeActivity needs to be recreated?

Sorry, I meant if the app is not backgrounded and HomeActivity needs to be recreated. I'm not sure this is possible without Don't Keep Activities in the current code base but writing code to protect against this (assuming it's not too time consuming) is valuable.

Reply:
ok I think i added all the changes we discussed.